vyper/ast/utils.py

```diff
@@ -8,7 +8,10 @@
 
 
 def parse_to_ast(
-    source_code: str, source_id: int = 0, contract_name: Optional[str] = None
+    source_code: str,
+    source_id: int = 0,
+    contract_name: Optional[str] = None,
+    add_fn_node: Optional[bool] = False,
```

Comment:
this is actually very clear

Reply:
i think the signature should be `add_fn_node: bool = False` though

or, `add_fn_node: Optional[str] = None`. that would allow you to set the "dummy_fn" name